src/main/java/htsjdk/samtools/CompressedIndexFileBuffer.java

```diff
@@ -0,0 +1,84 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.samtools.util.RuntimeIOException;
+
+import java.io.File;
+import java.io.IOException;
+
+
+class CompressedIndexFileBuffer implements IndexFileBuffer {
+
+    private BlockCompressedInputStream mCompressedStream;
+
+    CompressedIndexFileBuffer(File file) {
+        try {
+            mCompressedStream = new BlockCompressedInputStream(file);
+        } catch (IOException ioe) {
+            throw(new RuntimeIOException("Construction error of CSI compressed stream: " + ioe));
+        }
+    }
+
+    @Override
+    public void readBytes(final byte[] bytes) {
+        try {
+            mCompressedStream.read(bytes);
```

Comment:
I think there's a subtle bug here.   `mCompressedStream.read(bytes)` makes no guarantee that it will read as many bytes as you ask for.  You have to check the return value of read and continue reading until you either have read as many bytes as you require, or it returns -1 in which case you should probably throw.

Reply:
You need something like readFully or readBytes from BinaryCodec here I think.